src/flb_log.c

```diff
@@ -78,7 +151,21 @@ static inline int log_push(struct log_message *msg, struct flb_log *log)
             return write(STDERR_FILENO, msg->msg, msg->size);
         }
         ret = write(fd, msg->msg, msg->size);
+        /* Let's check the size before closing as we have open file descriptor. */
+        status = fstat(fd, &buf);
```

Comment:
Do we want to issue an `fstat` call each time a log line is pushed? I guess this is related to opening and closing the file each time as well which is something I am slightly against. If we decide that we'll keep on opening and closing the file each time I guess that overhead is acceptable, however, I'd really like us to change it if possible.
Of course keeping an internal file size counter could lead to slight deviations if the file was externally modified (which shouldn't really happen) but even then we could make a verification and re-sync call before rotating the file to catch external rotations if the user messes up the configuration (which we should communicate).

What do you think @edsiper?